osu.Desktop/OsuGameDesktop.cs

```diff
@@ -115,8 +116,10 @@ protected override UpdateManager CreateUpdateManager()
             }
         }
 
-        public override bool RestartAppWhenExited()
+        public override bool Restart()
         {
+            SessionStatics.SetValue(Static.RestartRequested, true);
```

Comment:
This is never set back to false if the user cancels the dialog on unsupported platforms.

With hold-to-confirm delay set to 0:

https://github.com/ppy/osu/assets/35318437/7f2ac4fe-ac37-49e5-9bc4-6ef09dc71e15